services/src/main/java/org/keycloak/cookie/SecureContextResolver.java

```diff
@@ -0,0 +1,32 @@
+package org.keycloak.cookie;
+
+import java.net.URI;
+
+class SecureContextResolver {
+
+    static boolean isSecureContext(URI uri) {
+        if (uri.getScheme().equals("https")) {
+            return true;
+        }
+
+        String host = uri.getHost();
+        if (host == null) {
+            return false;
+        }
+
+        if (host.equals("[::1]") || host.equals("[0000:0000:0000:0000:0000:0000:0000:0001]")) {
+            return true;
+        }
+
+        if (host.matches("127.0.0.\\d{1,3}")) {
```

Comment:
According to [the specification](https://w3c.github.io/webappsec-secure-contexts/#algorithms) only `127.0.0.1` should be considered a secure context.

Reply:
Nopes, that's incorrect, but it's also not correct in the PR, so should probably tune that.

`127.0.0.0/8` results to any IP in the range `127.0.0.1` to `127.255.255.254`

Reply:
Updated to work with the whole 127 block of IP addresses

Reply:
LGMT, let's resolve this